Front End Updates

angelocordon opened the pull request:
Set up front end architecture with Webpack; Vue can easily be installed in the future by running `docker-compose exec web rails webpack:install:vue`. 

Dockerfile and Docker Compose file is restructured to use a Docker Image that includes Ruby, Rails + NodeJS. This allows us to use Yarn as well against the Docker `web` service. Additionally, this includes a fix for the server.pid issue (#13).

Lastly, minimalistic updates to our welcome screen:

![image](https://user-images.githubusercontent.com/3891869/48960239-b9800680-ef1f-11e8-84af-f48380a4a55f.png)

angelocordon committed:
Basic restructure

angelocordon committed:
Mask tmp folder to deal with server.pid issue

Set up for tmp folder to be cleared out when running `docker-compose down` and
remove server.pid file - allowing Rails server to be booted up again normally
when Docker is restarted.

angelocordon committed:
Clean up Gemfile

Add Webpacker gem.
Remove sass-rails, uglifier and turbolinks with goals to moving all assets to
using Webpack instead of the Asset Pipeline.

angelocordon committed:
Install Webpacker

angelocordon committed:
Set packed application.js in view

angelocordon committed:
Clean up database yml

angelocordon committed:
Expose Webpack port in Docker

angelocordon committed:
Add sass-rails gem

Webpack SCSS compilaiton is slow, either through webpack config issues or others.
Adding the sass-rails gem will temporarily alleviate CSS pain points until we
can figure out the best move forward as ruby-sass will be deprecated in a few
months.

angelocordon committed:
Update layout views with stylesheet from asset pipepline

angelocordon committed:
Add Bulma

angelocordon committed:
Add basic UI with Bulma classes

angelocordon committed:
Update README

angelocordon committed:
Update Welcome page

stain88 commented on README.md:
````diff
@@ -19,7 +19,7 @@ git checkout develop
-docker-compose build && docker-compose up
+docker-compose up --build
 ```
 
````
Think we need something else here for `yarn`. I get

```
========================================
  Your Yarn packages are out of date!
  Please run `yarn install` to update.
========================================
```

angelocordon committed:
Update Docker setup

Move Webpack as its own service.
Set yarn integrity check based on environment variables.
Various folder names clean up.

angelocordon committed:
Update README

stain88 commented:
I'm happy with the current situation

jtkaufman737 commented:
Looking good to me as well.
